### ISSUE TITLE
ModuleNotFoundError Could not find module in path

### ISSUE THREAD

**Issue by tay1orjones** [codesandbox/codesandbox-client#4456]
# 🐛 bug report

## Description of the problem

I receive the error 
```
ModuleNotFoundError
Could not find module in path: 'carbon-addons-iot-react/es/node_modules/carbon-components-react/es/prop-types/types.js' relative to '/node_modules/carbon-addons-iot-react/es/src/components/Button/Button.jsx'
```
Despite this being available at the location indicated.

Here's a screencap of our build output for [`carbon-addons-iot-react`](https://github.com/IBM/carbon-addons-iot-react)

<img src="https://user-images.githubusercontent.com/3360588/85449669-a3a05080-b55d-11ea-8fa7-a185abc25dd3.png" width=300 />

Additionally you can see the file is available at this location in our package in unpkg
https://unpkg.com/carbon-addons-iot-react@2.93.5/es/node_modules/carbon-components-react/es/prop-types/types.js

## How has this issue affected you? What are you trying to accomplish?

This prevents us from using codesandbox as a bug reporting and example code tool.

### Your Environment

| Software         | Name/Version |
| ---------------- | ------------ |
| Сodesandbox      | 015cbaa7b (this was in the bottom left corner of the page) |
| Browser          | Chrome |
| Operating System | MacOS |

**Comment by tay1orjones** [comment]
@cowills @mhewson should #709 have fixed this? The error is still present in the linked sandbox above.

**Comment by cowills** [comment]
> @cowills @mhewson should #709 have fixed this? The error is still present in the linked sandbox above.

Nope, we just fixed the issue in our codebase by switching to a different bundle of our library.

**Comment by tay1orjones** [comment]
Ah! My mistake - I didn't realize the PR was in your codebase and not in codesandbox 🤦

**Comment by CompuIves** [comment]
Oh yes, this is interesting. The fact that it includes `/node_modules` confuses the bundler I think, and makes it look on unpkg to find whether the file exists... Doing a deeper dive now.

**Comment by CompuIves** [comment]
Yes, that's the case. The challenge here is that the in-browser bundler has an incomplete view on all files that are available, and because of this it has to make some assumptions when it can't find a file. If there's a file missing in `node_modules`, it tries to resolve a `package.json` and resolve the appropriate file from `unpkg`. I think that's where this goes wrong, because in this case it can't find the right version on unpkg.

**Comment by CompuIves** [comment]
The best solution would be to start saving paths for dependencies, as this would remove the whole guessing part. It would require us to regenerate the dependency cache for this, but other than that it seems like a sensible choice to me,

**Comment by CompuIves** [comment]
Keeping this one open since the saving paths feature is still something that's in the planning.

**Comment by apsamuel** [comment]
This is still an issue

**Comment by CompuIves** [comment]
Will reopen this one, and also rename the title a bit because we've narrowed the scope on this one. For what sandbox do you have this issue @apsamuel ?

**Comment by apsamuel** [comment]
> Will reopen this one, and also rename the title a bit because we've narrowed the scope on this one. For what sandbox do you have this issue @apsamuel ?

apologies for the delay - https://codesandbox.io/s/dme0w, I emailed out to codesandbox team and was informed there may have been an infrastructure issue on that specific occasion

**Comment by shaoster** [comment]
Willing to take any PRs for this? Looks like anything that transitively depends on socket.io hits this now: https://github.com/socketio/engine.io-client/blob/9eeed5e0c0a3a3598d27947dabc57b47ca3a8365/lib/transports/index.js#L1.

**Comment by CompuIves** [comment]
Interesting case, it looks like in the case you mentioned the require to XMLHTTPrequest is not in a node modules directory so it should work. But I should check closer, we can do that on Monday or Tuesday! cc @DeMoorJasper

**Comment by shaoster** [comment]
I think the `../..` prefix resolves to a node_modules rooted dir.

From the repro above: 
![image](https://user-images.githubusercontent.com/129236/119210179-fa1d5400-ba78-11eb-8b40-556b7581eade.png)

**Comment by jsjoeio** [comment]
I'm running into this issue with [this CodeSandbox](https://codesandbox.io/s/web-terminal-turorial-client-sgjgg)

Seems like some other folks are experiencing this and opened an issue in [socket-io-client](https://github.com/socketio/socket.io-client/issues/1494)

**Comment by CompuIves** [comment]
Right, that should not happen. I'll make sure that we take a look at this this week and come with a solution!
